core/src/executor/fetch.rs

```diff
@@ -60,3 +71,110 @@ pub async fn fetch<'a>(
 
     Ok(rows)
 }
+
+#[derive(futures_enum::Stream)]
+pub enum Rows<I1, I2> {
+    Derived(I1),
+    Table(I2),
+}
+pub async fn fetch_relation<'a>(
+    storage: &'a dyn GStore,
+    table_factor: &'a TableFactor,
+    filter_context: &Option<Rc<FilterContext<'a>>>,
+) -> Result<impl TryStream<Ok = Row, Error = Error, Item = Result<Row>> + 'a> {
+    match table_factor {
+        TableFactor::Derived { subquery, .. } => {
+            let filter_context = filter_context.as_ref().map(Rc::clone);
+            let rows = select(storage, subquery, filter_context).await?;
+
+            Ok(Rows::Derived(rows))
+        }
+        TableFactor::Table { name, .. } => {
+            let rows = storage
+                .scan_data(fetch_name(name)?)
+                .await?
+                .map_ok(|(_, row)| row);
+            let rows = stream::iter(rows);
+
+            Ok(Rows::Table(rows))
+        }
+    }
+}
+
+pub async fn fetch_join_columns<'a>(
```

Comment:
This looks that `fetch_join_columns` is currently only being used by `select/mod.rs`.
Then it would be better for `select/mod.rs` to have `fetch_join_columns` as private function.

Reply:
After implementing `fetch_relation_columns`, I realized that `fetch_relation_columns` and `fetch_join_columns` can refer each other to remove duplicated lines.
Now `fetch_join_columns` is used in both `select/mod.rs` and `fetch.rs` 😭

Reply:
Oh, I missed that point. Then current is right place to be 👍